### ISSUE TITLE
underbar doesn't work...

### ISSUE THREAD

**Issue by wm2015email** [KaTeX/KaTeX#2486]
Trying to write some matrix equation and "underbar" doesn't work... In my opinion, katex should support "underbar" out of the box without configuration...

My opinion is that underbar is much better than underline for marking vectors in a matrix equations because It's made for adding an underbar to a variable... whereas underline is for underlining text and it bleeds into the next character, which is veru annoy for matrix equations....

Example:
```
$E\Big[\big(\Theta- (E[\Theta]+\underbar{h}^TE[\underbar{X}]-\underbar{h}^T \underbar{X})\big)\underbar{X}\Big] = 0$
```
ok...I could use underline instead... but I hate underline because I always need to add spaces around the variables to get it to look good, even parenthesis, underlining a character next to a parenthesis just looks evil... and is too much work to fix by putting spaces everywhere... the alternative are equally annoying...vec and mathbf... they interfere too much with the use of hat and tilde... thus, the need to underbar to mark vectors....

The way I do it now:

```

$E\Big[\big(\Theta- (E[\Theta]+\underline{h}^TE[\underline{X}]-\underline{h}^T \underline{X})\big)\underline{X}\Big] = 0$
```

**Comment by ronkok** [comment]
> It's made for adding an underbar to a variable.

Actually, `\underbar` sets its contents to text mode. So (La)TeX will change the `h` in your example from italic to upright font.

If `\underbar` is really what you want, then the following macro will get you there:

    \def\underbar#1{\underline{\text{#1}}}

That's pretty easy, and I can add it to core KaTeX. Are you sure you want it?

**Comment by wm2015email** [comment]
Thanks... that would be very helpful for me because its more correctly formatted than underline for matrix equations...  If you fix it in KaTeX, then it propagates automatically to "math stack exchange" and typora were I need it... Thanks.

**Comment by edemaine** [comment]
Implemented in #2713.
